File: merge_sorted_array.py
def merge(nums1, m, nums2, n):
    nums1Copy = nums1[:m]

    i = 0
    j = 0
    k = 0

    while i < m and j < n:
        if nums1Copy[i] <= nums2[j]:
            nums1[k] = nums1Copy[i]
            i += 1
        else:
            nums1[k] = nums2[j]
            j += 1
        k += 1

    while i < m:
        nums1[k] = nums1Copy[i]
        i += 1
        k += 1

    while j < n:
        nums1[k] = nums2[j]
        j += 1
        k += 1

    nums1.sort()


def runTest(nums1, m, nums2, n, expected):
    nums1Copy = nums1.copy()

    merge(nums1, m, nums2, n)

    if nums1 == expected:
        print("pass")
    else:
        print("fail")
        print("Expected:", expected)
        print("Actual:", nums1Copy)


# Run the test cases
runTest([1, 2, 3, 0, 0, 0], 3, [2, 5, 6], 3, [1, 2, 2, 3, 5, 6])
runTest([1], 1, [], 0, [1])
runTest([0], 0, [1], 1, [1])
runTest([4, 5, 6, 0, 0, 0], 3, [1, 2, 3], 3, [1, 2, 3, 4, 5, 6])
runTest([4, 5, 6, 0, 0, 0], 3, [1, 2, 3], 3, [1, 2, 4, 5, 6, 6])

File: MergeSortedArray.js
const merge = (nums1, m, nums2, n) => {
    nums1.splice(m);
    nums2.splice(n);
    
    for(let i = 0; i < n; i++)
        nums1.push(nums2[i]);
    
    nums1.sort((a,b) => a-b);
};

const runTest = (nums1, m, nums2, n, expected) => {
    // Make a copy of nums1 to compare the result
    const nums1Copy = [...nums1];
    
    // Call the merge function
    merge(nums1, m, nums2, n);
    
    // Check if nums1 is equal to the expected result
    if (JSON.stringify(nums1) === JSON.stringify(expected)) {
        console.log("pass");
    } else {
        console.log("fail");
        console.log("Expected:", expected);
        console.log("Actual:", nums1Copy);
    }
}

// Run the test cases
runTest([1,2,3,0,0,0], 3, [2,5,6], 3, [1,2,2,3,5,6]);
runTest([1], 1, [], 0, [1]);
runTest([0], 0, [1], 1, [1]);
runTest([4,5,6,0,0,0], 3, [1,2,3], 3, [1,2,3,4,5,6]);
runTest([4, 5, 6, 0, 0, 0], 3, [1, 2, 3], 3, [1, 2, 4, 5, 6, 6]);